charmcraft/parts.py

```diff
@@ -237,9 +237,11 @@ def __init__(
         all_parts: Dict[str, Any],
         *,
         work_dir: pathlib.Path,
+        project_dir: pathlib.Path,
```

Comment:
can you document this, since it is right next to work_dir (add docstring), which I suspect answers my question in the general comment

Reply:
The parts processing work directory root is defined as an absolute path from the charm project directory (`charmdir/build`) if running in destructive mode, or as the user home directory if running on a managed environment.
```
        if env.is_charmcraft_running_in_managed_mode():
            work_dir = env.get_managed_environment_home_path()
        else:
            work_dir = self.buildpath
```

Reply:
Added parameter information to the `PartsLifecycle` docstring.